enclave_mgmt/model_creation.py

```diff
@@ -0,0 +1,414 @@
+        Enrollment.parse_obj(item)
+
+    return enrollments_df
+
+
+def users_model(clean_raw_df):
+    users_df = clean_raw_df['moodle_users']
+    users_df = users_df[['uuid', 'first_name', 'last_name', 'email']]
+
+    for item in users_df.to_dict(orient='records'):
+        User.parse_obj(item)
+
+    return users_df
+
+
+def assessments_and_grades_model(clean_raw_df):
+    grades_df = clean_raw_df['grades']
+    assessments_df = pd.DataFrame(
+        grades_df['assessment_name'].unique(), columns=['name']
+    )
+    assessments_df['id'] = assessments_df.index
+    grades_df = pd.merge(
+        grades_df, assessments_df, left_on='assessment_name', right_on='name'
+    )
+    grades_df.rename(columns={'id': 'assessment_id'}, inplace=True)
+    moodle_users = clean_raw_df['moodle_users'][['user_id', 'uuid']]
+    grades_df = pd.merge(grades_df, moodle_users, on='user_id')
+    grades_df.rename(columns={'uuid': 'user_uuid'}, inplace=True)
+    grades_df = grades_df[
+        ['assessment_id',
+         'user_uuid',
+         'course_id',
+         'grade_percentage',
+         'time_submitted'
+         ]
+    ]
+
+    def convert_percentage(x):
+        if x == '-':
+            return None
+        return float(x.strip("%"))
+
+    grades_df['grade_percentage'] = grades_df['grade_percentage'].map(
+        convert_percentage
+    )
+    grades_df = grades_df[grades_df['grade_percentage'].notnull()]
+    grades_df['time_submitted'] = grades_df['time_submitted'].astype(int)
+
+    for item in assessments_df.to_dict(orient='records'):
+        Assessment.parse_obj(item)
+    for item in grades_df.to_dict(orient='records'):
+        Grade.parse_obj(item)
+
+    return assessments_df, grades_df
+
+
+def ib_input_model(clean_raw_df):
+    ib_input_df = clean_raw_df['ib_input_instances']
+    ib_input_df = ib_input_df[
+                ['id',
+                 'content_id',
+                 'variant',
+                 'content',
+                 'prompt']]
+
+    for item in ib_input_df.to_dict(orient='records'):
+        InputInteractiveBlock.parse_obj(item)
+
+    return ib_input_df
+
+
+def ib_problem_model(clean_raw_df):
+    ib_problem_df = clean_raw_df['ib_pset_problems']
+    ib_problem_df = ib_problem_df[
+                  ['id',
+                   'content_id',
+                   'variant',
+                   'pset_id',
+                   'content',
+                   'problem_type',
+                   'solution',
+                   'solution_options']]
+
+    for item in ib_problem_df.to_dict(orient='records'):
+        ProblemSetProblem.parse_obj(item)
+
+    return ib_problem_df
+
+
+def course_contents_model(clean_raw_df):
+    course_contents_df = clean_raw_df['course_contents']
+    course_contents_df = course_contents_df[
+                  ['section',
+                   'activity_name',
+                   'lesson_page',
+                   'content_id',
+                   'visible'
```

Comment:
Per my other comments, let's back this addition out. I want to avoid any functional changes in this PR and limit it to just refactoring for existing schema definitions.